### PR TITLE
Feature/#24 receive signal from backend to frontend

### PR DESCRIPTION
**Pull request by jandegen**
Receiving a signal from the backend in the frontend and logging it to the output.
Signal is transmitted via UDP sockets to the NodeJS frontend server.

image

Test via from ./Apps:
```
docker-compose up --build
```

Changes:
- Running NodeJS within frontend container
- Sending extended test data from backend ( [x,y,z] tuples for 10 chanels concatinated to one byte array and sliced in frontend)
- Exposing ports:
   - Backend: 34254
   - Frontend: 34255

### CODE REVIEW

File: Apps/frontend/server/main.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+});
+
+server.on('listening', () => {
+  const address = server.address();
+  console.log(`Server listening on ${address.address}:${address.port}`);
+});
+
+server.bind(process.env.HOST_PORT, process.env.HOST_ADDRESS);
+
+function toArrayBuffer(buf) {
```
Comment (leandertolksdorf) [inline Apps/frontend/server/main.js:30]: unused function?

File: Apps/frontend/Dockerfile [hunk 2]
```diff
@@ -17,8 +17,12 @@ COPY . ./
 RUN npm run build
 
 # Use nginx for serving build
-FROM nginx:1.19-alpine
+FROM node:14-alpine AS serve
```
[the remark this answers is not in the document]
Comment (jandegen) [reply Apps/frontend/Dockerfile]: Thanks for pointing that out!
I implemented a temporary solution by starting nginx and node

File: Apps/frontend/server/main.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    const ab = new ArrayBuffer(buf.length);
+    const view = new Uint8Array(ab);
+    for (let i = 0; i < buf.length; ++i) {
+        view[i] = buf[i];
+    }
+    return ab;
+}
+
+function ab2str(buf) {
```
[the remark this answers is not in the document]
Comment (jandegen) [reply Apps/frontend/server/main.js]: Removed, artifact from buffer decoding

File: Apps/frontend/server/main.js (hunk 1 above)
Comment (jandegen) [reply Apps/frontend/server/main.js]: Removed, artifact from buffer decoding

**Review by leandertolksdorf — APPROVED** [review]

**Pull request merged**